### PR TITLE
Update use of 'JSON' to 'JSON::MaybeXS'

### PR DESCRIPTION
**Pull request by GuiltyDolphin**
More ***speed***!

IA Pages:

- https://duck.co/ia/view/cheat_sheets
- https://duck.co/ia/view/currency_in
- https://duck.co/ia/view/independence_day
- https://duck.co/ia/view/minecraft
- https://duck.co/ia/view/shruggie
- https://duck.co/ia/view/zodiac

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@GuiltyDolphin AFAIK JSON uses JSON::XS when available, so this shouldn't be necessary...

Can you double check on that?

**Comment by GuiltyDolphin** [comment]
@moollaza It uses `JSON::XS` when it can, but falls back to `JSON::PP` if that fails, if we're always going to be using `JSON::XS` though it kindah feels like we're introducing another step.

Your call (but I think we should either all use `JSON::XS` or all use `JSON`).

**Comment by GuiltyDolphin** [comment]
@moollaza *Maybe* this?

**Comment by moollaza** [comment]
Deploying to beta...

**Comment by GuiltyDolphin** [comment]
@moollaza That should be all of them!

**Comment by moollaza** [comment]
Deploying to beta...
/cc @zachthompson @jbarrett

File: lib/DDG/Goodie/Shruggie.pm [hunk 1]
```diff
@@ -10,8 +10,8 @@ package DDG::Goodie::Shruggie;
 
 use DDG::Goodie;
 use strict;
-use utf8; # needed to properly use the various unicode characters in the emoticons
-use JSON; # encoded the various other ASCII items in a JSON file
+use utf8;     # needed to properly use the various unicode characters in the emoticons
```
Comment (zachthompson) [inline lib/DDG/Goodie/Shruggie.pm]: white space

**Comment by zachthompson** [comment]
:+1:

**Comment by jbarrett** [comment]
We should be using `Cpanel::JSON::XS` wrapped via `JSON::MaybeXS` so there's a fallback.

`JSON::XS` and `JSON::PP` are effectively no longer compatible. While version pinning mitigates this to some extent, well, just look at the size of that warning box:

https://metacpan.org/pod/JSON#DESCRIPTION

In short, :+1:

**Pull request merged**